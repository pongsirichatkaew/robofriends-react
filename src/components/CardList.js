import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  console.log('render Card');
  // if (true) {
  //   throw new Error('Nooooooo');
  // }
  const cardComponent = robots.map((robot) => (
    <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} />
  ));
  return <div>{cardComponent}</div>;
};

export default CardList;
