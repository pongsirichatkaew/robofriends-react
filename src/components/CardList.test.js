import { shallow, mount, render } from 'enzyme';
import CardList from './CardList';
import React from 'react';
import toJson from 'enzyme-to-json';

describe('CardList test', () => {
  it('expect to render CardList Component', () => {
    const mockRobots = [
      {
        id: 1,
        name: 'John snow',
        email: 'john@gmail.com',
      },
    ];
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
  });
});
