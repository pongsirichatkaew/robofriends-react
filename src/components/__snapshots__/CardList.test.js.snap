// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList test expect to render CardList Component 1`] = `
<div>
  <Card
    email="john@gmail.com"
    id={1}
    key="1"
    name="John snow"
  />
</div>
`;
